### PR TITLE
Feature Toggles and Debug Symbols for Release Build

### PR DESCRIPTION
**Pull request by CPattar-NI**
Adding infrastructure to facilitate feature toggles. This will be a singleton class that reads from a config INI file, enabling and disabling a given feature.

Enabling the creation and storage of symbol files for release builds.

### CODE REVIEW

**Review by pratheekshasn — APPROVED** [review]

File: src/feature_toggles.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#include "feature_toggles.h"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <map>
+
+// Function to read feature configurations from an INI file
+void FeatureConfig::readConfigFromFile(const std::string& filePath) {
```
Comment (ni-sujain) [inline src/feature_toggles.cc:8]: Consider adding some documentation for the format of the ini file:

[section1]
key1=value1
key2=value2
[section2]
...

File: src/grpc_client.cc [hunk 2]
```diff
@@ -276,6 +277,9 @@ LIBRARY_EXPORT int32_t ClientUnaryCall(
     int32_t timeoutMs,
     grpc_labview::gRPCid* contextId)
 {
+    // Instantiating the feature toggles singleton that will read the feature configuration file
+    FeatureConfig::getInstance().readConfigFromFile("feature_config.ini");
```
Comment (ni-sujain) [inline src/grpc_client.cc:281]: Since we have already called this in InitCallbacks() do we need this here.

File: src/grpc_client.cc [hunk 3]
```diff
@@ -401,6 +405,9 @@ LIBRARY_EXPORT int32_t ClientBeginClientStreamingCall(
...
```
Comment (ni-sujain) [inline src/grpc_client.cc:409]: same comment as above

File: src/lv_interop.cc [hunk 4]
```diff
@@ -51,6 +52,9 @@ namespace grpc_labview
     //---------------------------------------------------------------------
     void InitCallbacks()
     {
+        // Instantiating the feature toggles singleton that will read the feature configuration file
+        FeatureConfig::getInstance().readConfigFromFile("feature_config.ini");
```
Comment (ni-sujain) [inline src/lv_interop.cc:56]: This seems to be a correct place to call readConfig . InitCallbacks() is the first method to get called while we do CreateClient, CreateServer, CreateSerializationSression etc.

File: src/grpc_server.cc [hunk 5]
```diff
@@ -178,6 +179,9 @@ namespace grpc_labview
         std::string serverKeyPath,
         ServerStartEventData *serverStarted)
     {
+        // Instantiating the feature toggles singleton that will read the feature configuration file
+        FeatureConfig::getInstance().readConfigFromFile("feature_config.ini");
```
Comment (ni-sujain) [inline src/grpc_server.cc:183]: same comment as above

**Review by ni-sujain — APPROVED** [review]

File: src/feature_toggles.cc (hunk 1 above)
Comment (CPattar-NI) [reply src/feature_toggles.cc:8]: Isn't this the generic INI format?

Comment (ni-sujain) [reply src/feature_toggles.cc:8]: yes

File: src/lv_interop.cc (hunk 4 above)
Comment (CPattar-NI) [reply src/lv_interop.cc:56]: Removed it in all other locations.

**Pull request merged**